core/src/main/kotlin/fr/sncf/osrd/api/api_v2/stdcm/STDCMEndpointV2.kt

```diff
@@ -173,16 +179,30 @@ class STDCMEndpointV2(private val infraManager: InfraManager) : Take {
     }
 }
 
-private fun parseSteps(infra: FullInfra, pathItems: List<STDCMPathItem>): List<STDCMStep> {
+private fun parseSteps(
+    infra: FullInfra,
+    pathItems: List<STDCMPathItem>,
+    startTime: ZonedDateTime
+): List<STDCMStep> {
     if (pathItems.last().stopDuration == null) {
         throw OSRDError(ErrorType.MissingLastSTDCMStop)
     }
+    if (pathItems.first().stepTimingData == null && pathItems.last().stepTimingData == null) {
+        throw OSRDError(ErrorType.MissingDepartureOrArrivalTime)
+    }
```

Comment:
We only need at least one value to be set, not necessarily first or last

Reply:
In the feature it's specified we should at least have the step timing data either the departure or arrival. Is that no longer needed?

Reply:
With the planned front-end we can expect to have either the start or end time, but we only *need* at least one time value on *any* step (including intermediate steps)